extensions/2.0/Vendor/EXT_mesh_primitive_edge_visibility/README.md

```diff
@@ -0,0 +1,150 @@
+- A [silhouette edge](https://en.wikipedia.org/wiki/Silhouette_edge) is any edge separating a front-facing triangle from a back-facing triangle. In Figure 1, one silhouette is visible along each of the curved left and right sides of the cylinder. Silhouette edges are *conditionally visible* - their visibility is determined at display time based on the orientations of the adjacent triangles relative to the camera. In Figure 2, each of the vertical edges encircling the cylinder represents a potential silhouette edge.
+- A hard edge is any edge attached to only a single triangle, or any edge between two logical faces of the 3D object. In Figure 1, hard edges are visible around the perimeters of the cylinder's circular end caps. Hard edges are *always visible* regardless of the camera direction.
+
+## Shortcomings of Existing Techniques
+
+Various techniques can be applied to a glTF asset to approximate the rendering in Figure 1. The approach illustrated in Figure 2 uses a wiremesh shader to render every edge of every triangle. While the interior edges can be useful for visualizing the structure of the triangle mesh, they obscure the underlying cylinder topology.
+
+<figure>
+<img src="./figures/wiremesh.png"/>
+<figcaption><em><b>Figure 2</b> Triangle edges drawn using a wiremesh technique</em></figcaption>
+</figure>
+
+Screen-space techniques (e.g., "toon" or "outline" shaders) can be applied to add edges during image post-processing. Because they operate on pixels rather than geometry, such techniques can only approximate the actual edges. The technique used in Figure 3, for example, fails to reconstruct the edges along the upper end cap of the cylinder where they fall inside the cylinder's volume projected onto the image plane.
+
+<figure>
+<img src="./figures/outline.png"/>
+<figcaption><em><b>Figure 3</b> Edges drawn using a screen-space technique</em></figcaption>
+</figure>
+
+The hard edges could be encoded explicitly into the glTF asset as additional primitives to be drawn together with the triangle mesh. However, this will generally produce a "stippling" effect where the edges and triangles collide in the depth buffer, as shown in Figure 4.
+
+<figure>
+<img src="./figures/depth-fighting.png"/>
+<figcaption><em><b>Figure 4</b> Edges drawn as separate primitives</em></figcaption>
+</figure>
+
+The [CESIUM_primitive_outline](https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Vendor/CESIUM_primitive_outline) extension provides an additional index buffer describing each hard edge as a line segment (a pair of indices into the triangle mesh's list of vertices). The extension leaves the details of how to render the edges up to the engine, requiring only that they render without depth-fighting. This approach satisfies the use case for which it was intended - displaying the edges of boxy, low-resolution buildings - but suffers some limitations:
```

Comment:
It would not need to be near-term, but is there potential that this extension could replace `CESIUM_primitive_outline` in the future?

Reply:
Yes, the hard edge visibility encoded by that extension is a subset of that encoded by this extension.

Reply:
@lilleyse if you agree, do you want to submit an issue to the `3d-tiles` repo for future consideration to replace `CESIUM_primitive_outline` with this extension over time?

Reply:
Yeah, makes sense to eventually replace `CESIUM_primitive_outline`. I opened https://github.com/CesiumGS/3d-tiles/issues/807.